kit/ChildSession.cpp

```diff
@@ -3146,6 +3233,33 @@ void ChildSession::loKitCallback(const int type, const std::string& payload)
         break;
     case LOK_CALLBACK_EXPORT_FILE:
     {
+        bool isAbort = payload == "ABORT";
+        bool isPending = payload == "PENDING";
+
+        if (isPending)
+        {
+            sendTextFrame("blockui: ");
+            return;
+        }
+
+        sendTextFrame("unblockui: ");
+
+        if (isAbort)
+        {
+            _exportAsWopiUrl = "";
+            return;
+        }
+        else if (!_exportAsWopiUrl.empty())
+        {
+            std::string encodedURL;
+            Poco::URI::encode(payload, "", encodedURL);
+
+            sendTextFrame("exportas: url=" + encodedURL + " filename=" + _exportAsWopiUrl);
```

Comment:
What's the purpose of sending this? We also send it when the upload is complete, which looks like we are sending this message twice, once on exporting the document on the server (locally) and once when we upload to storage.

This would seem wrong, since the client can't do anything with the exported file on the server (locally) and can only possibly do something (like reloading) when the file is uploaded.

Maybe I'm missing the purpose of these "exportas" messages.

Reply:
at this point file was exported locally, now we need to send the message so we will start the uploading part